src/Symfony/Component/Security/Core/Authorization/Voter/AbstractVoter.php

```diff
@@ -107,7 +120,32 @@ public function vote(TokenInterface $token, $object, array $attributes)
      * @param object               $object
      * @param UserInterface|string $user
      *
+     * @deprecated This method will be removed in 3.0 - override voteOnAttribute instead.
+     *
      * @return bool
      */
-    abstract protected function isGranted($attribute, $object, $user = null);
+    protected function isGranted($attribute, $object, $user = null)
+    {
```

Comment:
this method should trigger a deprecation warning when being called, in case someone implements ``voteOnAttribute`` by calling it.